ubuntu-advantage-client/azure_cleanup.py

```diff
@@ -0,0 +1,109 @@
+
+# Copyright 2020 Canonical Ltd.
+# Lucas Moura <lucas.moura@canonical.com>
+import os
+import json
+
+from pycloudlib.azure.util import get_client
+from azure.mgmt.resource import ResourceManagementClient
+
+import argparse
+
+
+CI_DEFAULT_TAG = "uaclient"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-t", "--tag", dest="tag", action="store", default=CI_DEFAULT_TAG,
+        help=(
+            "Tag used as a prefix to search for resources to be deleted."
+            "Default: {}".format(CI_DEFAULT_TAG))
+    )
+    parser.add_argument(
+        "--client-id", dest="client_id", 
+        help="Client id used to access azure api"
+    )
+    parser.add_argument(
+        "--client-secret", dest="client_secret", 
+        help="Client secret used to access azure api"
+    )
+    parser.add_argument(
+        "--tenant-id", dest="tenant_id", 
+        help="Tenant id used to access azure api"
+    )
+    parser.add_argument(
+        "--subscription-id", dest="subscription_id", 
+        help="Subscription id used to access azure api"
+    )
+    parser.add_argument(
+        "--credentials-path", dest="credentials_path",
+        help="""
+            Path where the credentials file is stored. That file must a be
+            a json dict containing all the necessary credentials to manage
+            Azure resources. To successfuly be used in this script,
+            the json file must have the following keys:  client_id,
+            client_secret, tenant_id, subscription_id.
+            """
+        )
+
+    return parser.parse_args()
+
+
+def clean_azure(tag, client_id, client_secret, tenant_id, subscription_id):
+    """Clean up all running Azure resources and resource groups"""
+    config_dict = {
+        "clientId": client_id,
+        "clientSecret": client_secret,
+        "tenantId": tenant_id,
+        "subscriptionId": subscription_id
+    }
+
+    resource_client = get_client(
+        ResourceManagementClient, config_dict
+    )
+
+    print('# searching for resource groups matching tag {}'.format(tag))
+    for resource_group in resource_client.resource_groups.list():
+        tags = resource_group.tags
+
+        if tags:
+            for tag_value in tags.values():
+                if tag_value.startswith(tag):
+                    resource_group_name = resource_group.name
+                    print('# deleting resource group: {}'.format(
+                        resource_group_name))
+                    result = resource_client.resource_groups.delete(
+                        resource_group_name=resource_group_name
+                    )
+                    result.wait()
+                    break
+
+
+def load_azure_config(credentials_path):
+    with open(credentials_path, 'r') as f:
+        return json.load(f)
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    if args.credentials_path:
+        if os.path.exists(args.credentials_path):
```

Comment:
I generally prefer if a script handles the error of a path that does not exist directly but I guess for this purpose it's low risk.

Reply:
This is just an attempt to fallback to the raw credentials if necessary. But thinking of the error we will generate, it will not be clear for an user that the script failed because the path does not exist. I will update the script to raise an Exception if the file does not exists